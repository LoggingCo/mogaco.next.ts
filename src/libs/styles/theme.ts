import { DefaultTheme } from 'styled-components';

const palette = {
  primary: {
    100: '#BFDEEE',
    200: '#9DCDE7',
    300: '#3AA5DD',
  },
  subColor: '#EBF1F4',
  fontColor: '#222222',
  fontSubColor: '#ffffff',
  error: '#EB5757',
  subError: '#FFBA52',
  gray: {
    100: '#e9e9e9',
    200: '#D9D9D9',
    300: '#656565',
  },
  yellow: '#FFE381',
  puple: '#EDC4FC',
  sky: '#A4EEFF',
} as const;

const fontSize = {
  small: '14px',
  medium: '16px',
  large: '18px',
  xLarge: '24px',
  xxLarge: '32px',
} as const;

const fontWeight = {
  thin: '100',
  regular: '400',
  medium: '500',
  bold: '700',
} as const;

const lineHeight = {
  small: '16px',
  medium: '20px',
  large: '24px',
  xLarge: '28px',
} as const;

export type PaletteTypes = typeof palette;
export type PaletteKeyTypes = keyof typeof palette;
export type FontSizeTypes = typeof fontSize;
export type FontWeightTypes = typeof fontWeight;
export type LineHeightTypes = typeof lineHeight;

const theme: DefaultTheme = {
  palette,
  fontSize,
  fontWeight,
  lineHeight,
};

export default theme;
