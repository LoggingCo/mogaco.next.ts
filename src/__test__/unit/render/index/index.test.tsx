import { render, screen } from '@testing-library/react';
import Home from '@/pages/index';
import { ThemeProvider } from 'styled-components';
import theme from '@/libs/styles/theme';
import userEvent from '@testing-library/user-event';
import RoomListPage from '@/pages/room-list';
import { RecoilRoot } from 'recoil';

describe('<Home />', () => {
  it('홈페이지가 렌더링 되고 스냅샷을 비교한다', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Home />
      </ThemeProvider>,
    );
    const home = screen.getByText('mogaco');

    expect(home).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('go 버튼 클릭 시 room-list 페이지로 이동된다', () => {
    render(
      <ThemeProvider theme={theme}>
        <Home />
      </ThemeProvider>,
    );

    const $goButton = screen.getByText('go');
    userEvent.click($goButton);

    render(
      <ThemeProvider theme={theme}>
        <RecoilRoot>
          <RoomListPage />
        </RecoilRoot>
      </ThemeProvider>,
    );

    const $roomListPage = screen.getByPlaceholderText('모임 이름 입력 (최대 10자)');
    expect($roomListPage).toBeInTheDocument();
  });
});
