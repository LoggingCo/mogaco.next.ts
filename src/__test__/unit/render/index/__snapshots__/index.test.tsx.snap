// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders a heading 1`] = `
<div>
  <div
    class="sc-hLBbgP eFmqOR"
  >
    <div
      class="sc-dkrFOg jcNiDM"
    >
      <a
        href="/"
      >
        mogaco
      </a>
    </div>
    <div
      class="sc-eDvSVe kjKcFA"
    >
      Mogaco 
      {
      <div
        class="Typist "
      >
        <span
          class="Cursor Cursor--blinking"
        >
          |
        </span>
      </div>
      }
      <button>
        go
      </button>
    </div>
  </div>
</div>
`;

exports[`<Home /> 홈페이지가 렌더링 되고 스냅샷을 비교한다 1`] = `
<div>
  <div
    class="sc-hLBbgP eFmqOR"
  >
    <div
      class="sc-dkrFOg jcNiDM"
    >
      <a
        href="/"
      >
        mogaco
      </a>
    </div>
    <div
      class="sc-eDvSVe kjKcFA"
    >
      Mogaco 
      {
      <div
        class="Typist "
      >
        <span
          class="Cursor Cursor--blinking"
        >
          |
        </span>
      </div>
      }
      <button>
        go
      </button>
    </div>
  </div>
</div>
`;

exports[`<Home /> 홈페이지가 렌더링 된다 1`] = `
<div>
  <div
    class="sc-hLBbgP eFmqOR"
  >
    <div
      class="sc-dkrFOg jcNiDM"
    >
      <a
        href="/"
      >
        mogaco
      </a>
    </div>
    <div
      class="sc-eDvSVe kjKcFA"
    >
      Mogaco 
      {
      <div
        class="Typist "
      >
        <span
          class="Cursor Cursor--blinking"
        >
          |
        </span>
      </div>
      }
      <button>
        go
      </button>
    </div>
  </div>
</div>
`;
